src/lib/support/BitFlags.h

```diff
@@ -102,7 +102,7 @@ class BitFlags
      */
     BitFlags & Clear(const BitFlags & other)
     {
-        mValue &= ~other.mValue;
+        mValue &= static_cast<IntegerType>(~static_cast<IntegerType>(other.mValue));
```

Comment:
This is probably not right.  The right thing is:
```
mValue = static_cast<IntegerType>(mValue & ~other.mValue);
```
or so.  Because using `&=` causes the result of the `&` (which may have been promoted to a different type) to be assigned to mValue...

Reply:
could you provide a test showing this behaviour?

We relied a lot on unit tests to prove that bitflags works for us as expected.

Reply:
It _works_ correctly.  When it compiles.  It fails to compile on some compilers, I expect.  Depending on optimization settings and the like.

Reply:
Can we have a compile variant test for this? When this was fixed, we added a unit test that validated we compile.